protocol/contracts/admin/TempleTeamPaymentsFactory.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/proxy/Clones.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "./TempleTeamPaymentsV2.sol";
+
+contract TempleTeamPaymentsFactory is Ownable {
+    struct FundingData {
+        address paymentContract;
+        uint256 totalFunding;
+        uint16 epoch;
+    }
+
+    address public templeTeamPaymentsImplementation;
+    uint16 public initialEpoch;
+    uint16 public lastPaidEpoch;
+    mapping(uint256 => FundingData) public epochsFunded;
+
+    event FundingPaid(
+        address paymentToken,
+        uint256 indexed fundingRound,
+        address[] indexed dests,
+        uint256[] indexed amounts
+    );
+    event FundingDeployed(
+        address paymentToken,
+        uint256 indexed fundingRound,
+        address[] indexed dests,
+        uint256[] indexed amounts,
+        address deployedTo
+    );
+
+    constructor(uint16 _lastPaidEpoch) {
+        templeTeamPaymentsImplementation = address(new TempleTeamPaymentsV2());
+        lastPaidEpoch = _lastPaidEpoch;
+        initialEpoch = _lastPaidEpoch + 1;
+    }
+
```

Comment:
I would add a function to the contract to set a new implementation contract if an upgrade/changes happen.